bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/PerChannelBookieClient.java

```diff
@@ -1180,6 +1180,7 @@ private void writeAndFlush(final Channel channel,
         if (channel == null) {
             LOG.warn("Operation {} failed: channel == null", StringUtils.requestToString(request));
             errorOut(key);
+            ReferenceCountUtil.release(request);
```

Comment:
I think it's the wrong thing to do generally in this method. I'll share a new PR where I explain an alternative approach that handles both the V3 protocol and the UnsafeByteOperations issue and the issue that was found.

Reply:
To trace the context easier, I move the second comment here.

> I think it's the wrong thing to do generally in this method. I'll share a new PR where I explain an alternative approach that handles both the V3 protocol and the UnsafeByteOperations issue and the issue that was found.

> I don't think that the current changes in this PR are ok. I created https://github.com/apache/bookkeeper/pull/4293 to show what I think that should be done.

I added a section named `Explanation` in the Motivation, it explains that the current change is correct. Thanks for mentioning me ❤️